src/connection/ConnectionModule.ts

```diff
@@ -0,0 +1,91 @@
+const notImplementedError = (functionName: string): never => {
+  throw new Error(`Not Implemented: ${functionName}`);
+};
+
+export abstract class ConnectionModule {
```

Comment:
We need to resync with the changes in https://github.com/neo4j-labs/neodash/pull/767.
Lets discuss on a separate thread.